vimeo-networking/src/main/java/com/vimeo/networking/model/Video.java

```diff
@@ -381,19 +382,40 @@ public Play getPlay() {
     // VOD
     // -----------------------------------------------------------------------------------------------------
     // <editor-fold desc="VOD">
+
+    /**
+     * A VOD video can have multiple purchase types active at once, this is a convenience method that
+     * picks one of them based on the following priority:
+     * 1) trailer
+     * 2) purchased
+     * 3) both rental and subscription? choose the later expiration, expirations equal? choose subscription
+     * 4) subscription
+     * 5) rental
+     *
+     * @return the VodVideoType of the video or {@code VodVideoType.NONE} if it is not a VOD video or
+     * {@code VodVideoType.UNKNOWN} if it is a VOD video that is not marked as rented, subscribed or bought
+     */
     public VodVideoType getVodVideoType() {
         if (isVod()) {
             if (isTrailer()) {
                 return VodVideoType.TRAILER;
             }
-            if (isRental()) {
-                return VodVideoType.RENTAL;
+            if (isPurchase()) {
+                return VodVideoType.PURCHASE;
+            }
+            // rentals or subscriptions that have been purchased will always have an expiration date
+            Date rentalExpires = getRentalExpiration();
+            Date subscriptionExpires = getSubscriptionExpiration();
+            if (rentalExpires != null && subscriptionExpires != null) {
```

Comment:
Could there be a case where only one of these dates is null? or would that indicate a problem with the API?